allensdk/brain_observatory/behavior/internal/behavior_base.py

```diff
@@ -0,0 +1,130 @@
+import abc
+
+from typing import Dict, NamedTuple
+
+import numpy as np
+import pandas as pd
+
+
+RunningSpeed = NamedTuple("RunningSpeed", [("timestamps", np.ndarray),
+                                           ("values", np.ndarray)])
+
+
+class BehaviorBase(abc.ABC):
+    """Abstract base class implementing required methods for interacting with
+    behavior session data.
+
+    Child classes should be instantiated with a fetch API that implements these
+    methods. Both fetch API and session object should inherit from this base.
+    """
+    @abc.abstractmethod
+    def get_licks(self) -> pd.DataFrame:
+        """Get lick data from pkl file.
+
+        Returns
+        -------
+        np.ndarray
+            A dataframe containing lick timestamps.
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def get_rewards(self) -> pd.DataFrame:
+        """Get reward data from pkl file.
+
+        Returns
+        -------
+        pd.DataFrame
+            A dataframe containing timestamps of delivered rewards.
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def get_running_data_df(self) -> pd.DataFrame:
+        """Get running speed data.
+
+        Returns
+        -------
+        pd.DataFrame
```

Comment:
investigating this might be out of scope for now, but this would be a good place to document these "various signals".